### PR TITLE
Fix CI workflows to always run

### PR DESCRIPTION
**Pull request by jezdez**
### Description

It seems as if the work in #10864 in combination with #11286 might have created a scenario in which tests were skipped for PR containing docs.

### Checklist - did you ...

- [ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -252,7 +249,13 @@ jobs:
...
+
       - name: Test Summary
         uses: test-summary/action@v1
         with:
           paths: ./test-results-${{ github.sha }}-**/test-report*.xml
+
+      - name: Decide whether the needed jobs succeeded or failed
+        uses: re-actors/alls-green@release/v1
+        with:
+          jobs: ${{ toJSON(needs) }}
```
Comment (kenodegard) [inline .github/workflows/ci.yml:261]: This will always be true since the `needs` condition will prevent the `analyze` job from running if anything failed.

Comment (jezdez) [reply .github/workflows/ci.yml:261]: Even with the `if: always()` condition above?

Comment (jezdez) [reply .github/workflows/ci.yml:261]: I've following the example on https://github.com/re-actors/alls-green#usage, e.g. https://github.com/pyca/cryptography/blob/main/.github/workflows/ci.yml#L518-L531

Comment (jezdez) [reply .github/workflows/ci.yml:261]: https://github.com/orgs/community/discussions/26733#discussioncomment-3253155 has some background

Comment (kenodegard) [reply .github/workflows/ci.yml:261]: oh I see, the analysis job is always run, https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-not-requiring-successful-dependent-jobs

so we want the analysis job to run but "fail" if any of the dependencies also failed?

Comment (jezdez) [reply .github/workflows/ci.yml:261]: Yeah, basically

**Review by kenodegard — APPROVED** [review]

**Pull request merged**